pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -423,19 +423,26 @@ def _map_tool_definition(f: ToolDefinition) -> BetaToolParam:
 
 
 def _map_usage(message: BetaMessage | BetaRawMessageStreamEvent) -> usage.Usage:
-    if isinstance(message, BetaMessage):
-        response_usage = message.usage
-    elif isinstance(message, BetaRawMessageStartEvent):
-        response_usage = message.message.usage
-    elif isinstance(message, BetaRawMessageDeltaEvent):
-        response_usage = message.usage
+    """Maps Anthropic API message object to pydantic-ai Usage object, extracting integer-type usage statistics.
+
+    Handles BetaMessage, BetaRawMessageStartEvent, BetaRawMessageDeltaEvent.
+    Returns empty Usage if type doesn't contain usage info.
+    """
+    msg_type = type(message)
+
+    if msg_type is BetaMessage:
+        response_usage = cast(BetaMessage, message).usage
+    elif msg_type is BetaRawMessageStartEvent:
+        response_usage = cast(BetaRawMessageStartEvent, message).message.usage
+    elif msg_type is BetaRawMessageDeltaEvent:
+        response_usage = cast(BetaRawMessageDeltaEvent, message).usage
     else:
         # No usage information provided in:
         # - RawMessageStopEvent
         # - RawContentBlockStartEvent
         # - RawContentBlockDeltaEvent
         # - RawContentBlockStopEvent
-        return usage.Usage()
+        return _EMPTY_USAGE
```

Comment:
doesn't this mean that if someone edits the returned usage, all "empty usage" going forward will use that usage!

Reply:
thanks, I updated this change with this fix. The new change should still be around 100% faster